### PR TITLE
Add sys.excepthook to log errors and quit cleanly

### PR DESCRIPTION
**Pull request by inglesp**
This should make debugging things like #231 simpler.

### CODE REVIEW

File: mu/app.py [hunk 1]
```diff
@@ -42,12 +42,21 @@ def setup_logging():
...
+    sys.exit(1)
+
+
 def run():
     """
     Creates all the top-level assets for the application, sets things up and
     then runs the application.
     """
     setup_logging()
+    sys.excepthook = excepthook
```
Comment (carlosperate) [inline mu/app.py]: Just a suggestion, we could move this line into `setup_logging()`.

File: mu/app.py [hunk 2]
```diff
@@ -42,12 +42,21 @@ def setup_logging():
     print('Logging to {}'.format(LOG_FILE))
 
 
+def excepthook(*exc_args):
+    """
+    Log exception and exit cleanly.
+    """
+    logging.error('Unrecoverable error', exc_info=(exc_args))
```
Comment (carlosperate) [inline mu/app.py]: Might be useful to also propagate the exception through the normal route, that way we can still see the trace from running from the python source or executable command line:
```
sys.__excepthook__(exc_args)
```

**Comment by carlosperate** [comment]
A bit unrelated to the PR itself (which LGTM), but for the sake of understanding #231 a bit better:

> This should make debugging things like #231 simpler.

I thought that issue caused a sudden crash and didn't get any Python exception raised, or it would have printed it to the console, no?

**Comment by inglesp** [comment]
> I thought that issue caused a sudden crash and didn't get any Python exception raised, or it would have printed it to the console, no?

The exception does get written to stdout, but where does stdout go for an application launched from Finder?

**Comment by carlosperate** [comment]
Right, I forgot it worked correctly when launched from the terminal.

**Pull request merged**